### ISSUE TITLE
NoMethodError: undefined method `+' for nil:NilClass

### ISSUE THREAD

**Issue by martnst** [jfelchner/ruby-progressbar#114]
We use `ruby-progressbar` in combination with the [parallel gem](https://github.com/grosser/parallel). It runs in 250 threads doing HTTP Head requests to check for updates of a bunch of files. Sometime it crashes with `NoMethodError: undefined method`+' for nil:NilClass` in this line https://github.com/jfelchner/ruby-progressbar/blob/master/lib/ruby-progressbar/output.rb#L51

**Comment by jfelchner** [comment]
@martnst sorry for the delay on this reply!

I'm not entirely sure here but since you've got the setup where the issue is happening, I'd say it would probably be easiest for you to repro and send a PR.

Unfortunately concurrency is a hard problem.  We have [similar issues in fuubar](https://github.com/thekompanee/fuubar/issues/22) (which uses ruby-progressbar) in using the parallel-tests gem.  No one has been interested in taking that up either.

So bottom line is, I'd love to get a PR for this, but I'm personally not going to invest the time in tracking this down.
